browser/ui/browser_commands.cc

```diff
@@ -898,6 +901,56 @@ void ScrollTabToBottom(Browser* browser) {
   contents->ScrollToBottomOfDocument();
 }
 
+namespace {
+
+class BookmarksExportListener : public ui::SelectFileDialog::Listener {
+ private:
+  raw_ptr<Profile> profile_;
+
+ public:
+  scoped_refptr<ui::SelectFileDialog> fileSelector;
+
+  explicit BookmarksExportListener(Profile* profile) : profile_(profile) {}
+  void FileSelected(const ui::SelectedFileInfo& file, int index) override {
+    bookmark_html_writer::WriteBookmarks(profile_, file.file_path, nullptr);
+    std::cout << "Bookmarks exported to " << file.file_path.value()
+              << std::endl;
+
+    delete this;
+  }
+  void FileSelectionCanceled() override {
+    std::cout << "File selection for bookmarks export canceled" << std::endl;
+    delete this;
+  }
+};
+
+}  // namespace
+
+void ExportAllBookmarks(Browser* browser) {
+  const std::time_t t = std::time(0);
+  const std::tm* time = std::localtime(&t);
+  const std::string year = std::to_string(time->tm_year + 1900);
+  const std::string month = std::to_string(time->tm_mon + 1);
+  const std::string day = std::to_string(time->tm_mday);
+  const std::string formatted_time = year + "_" + month + "_" + day;
+  const std::string defaultBookmarksFilename = base::StringPrintf(
+      "%s_Brave_browser_bookmarks.html", formatted_time.c_str());
+
+  ui::SelectFileDialog::FileTypeInfo file_types;
+
+  // Only show HTML files in the file dialog.
+  file_types.extensions.push_back({"html"});
+
+  BookmarksExportListener* listener =
+      new BookmarksExportListener(browser->profile());
+
+  listener->fileSelector = ui::SelectFileDialog::Create(listener, nullptr);
+  listener->fileSelector->SelectFile(
+      ui::SelectFileDialog::SELECT_SAVEAS_FILE, u"Export Bookmarks",
```

Comment:
from `ui/base/l10n/l10n_util.h`

I think the string is in `chrome/grit/bookmarks_resources.h`

```suggestion
      ui::SelectFileDialog::SELECT_SAVEAS_FILE, l10n_util::GetStringUTF16(IDS_BOOKMARK_MANAGER_MENU_EXPORT),
```